protocol-units/da-sequencer/node/src/storage/mod.rs

```diff
@@ -45,13 +44,13 @@ impl TxCompositeKey {
 	}
 
 	/// Decode a composite key from a byte slice.
-	fn decode(bytes: &[u8]) -> Result<Self, DaSequencerError> {
+	fn _decode(bytes: &[u8]) -> Result<Self, DaSequencerError> {
 		if bytes.len() != 44 {
 			return Err(DaSequencerError::StorageFormat("Invalid composite key length".into()));
 		}
 
-		let timestamp = u64::from_be_bytes(bytes[0..8].try_into().unwrap());
-		let index_in_batch = u32::from_be_bytes(bytes[8..12].try_into().unwrap());
+		let timestamp = u64::from_be_bytes(bytes[0..8].try_into()?);
+		let index_in_batch = u32::from_be_bytes(bytes[8..12].try_into()?);
```

Comment:
Ah yes, not show how these unwrap calls got merged, thanks.

Reply:
It has been delivered courtesy of RustRover. :)

Reply:
They have been merged because they are tested just before, so panic won't ever happen.